org.adempiere.base/src/org/compiere/model/MLookupFactory.java

```diff
@@ -516,10 +516,13 @@ static private MLookupInfo getLookup_Table (Properties ctx, Language language,
 				}
 			}
 			realSQL.append(",").append(TableName).append(".IsActive");
+
+			String realKeyColumn = TableName + (MTable.get(ctx, TableName).isUUIDKeyTable() ? "_UU" : "_ID");

```

Comment:
Should use `PO.getUUIDColumnName(tableName)` to get UU column name. Also, you don't have to do this at all if KeyColumn is already ends with _ID or _UU.

Reply:
@nmicoud / @hengsin 
Please check my comment at https://idempiere.atlassian.net/browse/IDEMPIERE-6365?focusedCommentId=51426

Reply:
change done